crates/biome_css_formatter/src/css/lists/relative_selector_list.rs

```diff
@@ -23,7 +23,27 @@ impl FormatRule<CssRelativeSelectorList> for FormatCssRelativeSelectorList {
             // For example, a selector like `div span a` is structured like
             // `[div, [span, [a]]]`, so `a` would end up double-indented if it
             // was handled by the selector rather than here.
-            joiner.entry(&group(&indent(&formatted)));
+            let mut has_comments_before = false;
+            if let Some(relative_selector) = formatted.node()?.as_css_relative_selector() {
+                if let Some(computed_selector) =
+                    relative_selector.selector()?.as_css_compound_selector()
+                {
+                    if let Some(simple_selector) = computed_selector.simple_selector() {
+                        if let Some(type_selector) = simple_selector.as_css_type_selector() {
+                            let value_token = type_selector.ident()?.value_token()?;
+                            if value_token.has_leading_comments() {
+                                has_comments_before = true;
+                            }
+                        }
+                    }
+                }
+            }
+
```

Comment:
You don't need all these `let/if`, just use `.map`/`and_then` and `unwrap_or_default` at the end

Reply:
I think you can use `transpose` to solve the issue. Here's an example used inside the formatter ([`Option` function](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) and [`Result` function](https://doc.rust-lang.org/std/result/enum.Result.html#method.transpose))

https://github.com/biomejs/biome/blob/4713c5263dce30eafdd05c700e2503bf15ef4bd3/crates/biome_js_formatter/src/utils/assignment_like.rs#L1321-L1325

Reply:
it seems the `transpose` api deal `Option<Result<XX>>` into `Result<Option<XX>>`

![image](https://github.com/user-attachments/assets/a8a01cd2-4f12-4ffc-8e4e-ea3cbafa489b)

Maybe I need an api to do the opposite here

Reply:
I used your code that you pasted in your second comment, and **the compiler** told me how to fix it :) 

Sometimes, a good old `cargo c` does the trick! 

I committed the code here [`4d1f892` (#4392)](https://github.com/biomejs/biome/pull/4392/commits/4d1f892b31a89eec8f4c670c4c91fe3132dfac7a)

Reply:
Also, I renamed the variable. We don't use "before" and "after", but we tend to use the technical names: "leading" and "trailing"

Reply:
ok，thanks～